app/jobs/browse_everything_ingest_job.rb

```diff
@@ -8,13 +8,73 @@ def perform(resource_id, controller_scope_string, pending_upload_ids)
     controller_scope = controller_scope_string.constantize
     change_set_persister = controller_scope.change_set_persister
     change_set_class = controller_scope.change_set_class
-    pending_upload_ids = pending_upload_ids.map { |x| Valkyrie::Types::ID[x] }
+    resource = change_set_persister.metadata_adapter.query_service.find_by(id: Valkyrie::ID.new(resource_id))
+    pending_uploads = resource.pending_uploads.select { |upload| pending_upload_ids.include?(upload.id.to_s) }
+
+    upload_ids = []
+    pending_uploads.each do |upload|
+      if upload.container?
+
+        file_attributes = {
+          id: upload.local_id.first,
+          container: upload.container?,
+          provider: upload.provider.first
+        }
+
+        unless upload.auth_token.nil?
+          file_attributes[:auth_token] = upload.auth_token.first
+          file_attributes[:auth_header] = upload.auth_header
+        end
+
+        retriever = BrowseEverything::Retriever.new
+        member_resources = retriever.member_resources(file_attributes)
+        provider = BrowseEverything::Retriever.build_provider(upload.provider.first)
+        # Create the new uploads
+        member_uploads = member_resources.map do |member_file|
+          member_attributes = {
+            local_id: member_file[:id],
+            file_name: member_file[:file_name],
+            file_size: member_file[:file_size],
+            url: member_file[:url],
+            type: member_file[:container] ? "container" : "file",
+            provider: member_file[:provider]
+          }
+
+          if member_file.key?(:auth_token)
+            # This uses the provider constructed for the parent container
+            file_auth_header = provider.class.authorization_header(member_file[:auth_token])
+
+            member_attributes[:auth_token] = member_file[:auth_token]
+            member_attributes[:auth_header] = file_auth_header
+          end
+
+          PendingUpload.new(
+            member_attributes.merge(
+              id: SecureRandom.uuid,
+              created_at: Time.current.utc.iso8601
+            )
+          )
+        end
+
+        # Persist the new uploads
+        change_set_persister.buffer_into_index do |buffered_change_set_persister|
+          change_set = change_set_class.new(resource)
+          change_set.validate(pending_uploads: change_set.pending_uploads + member_uploads)
```

Comment:
One aspect of this where I am still struggling is how to delete the pending uploads for directories (they are added as `PendingUpload` objects, but will never be appended as `FileSets` and should be deleted from the `Resource`).

Reply:
I attempted to address this by using `remaining_uploads` below.